src/coll/mod.rs

```diff
@@ -111,9 +118,25 @@ impl Collection {
                 read_concern,
                 write_concern,
             }),
+            _phantom: Default::default(),
         }
     }
 
+    /// Gets a clone of the `Collection` with a different type `U`.
+    pub fn clone_with_type<U>(&self) -> Collection<U>
+    where
+        U: Serialize,
+    {
+        let mut options = CollectionOptions::builder().build();
+        options.selection_criteria = self.inner.selection_criteria.clone();
+        options.read_concern = self.inner.read_concern.clone();
+        options.write_concern = self.inner.write_concern.clone();
+
+        self.inner
```

Comment:
Interesting! Another way to work around this would be to give a separate generic parameter to the `new` method, e.g. `fn new<U: Serialize>(...) -> Collection<U>`. That way, it wouldn't be tied to returning the same generic parameter.

Reply:
oh good idea, i'll update